kubeconfig_plugin.py

```diff
@@ -111,63 +111,151 @@ class ErrorOutput:
+        if "clusters" not in kubeconfig:
+            return "error", ErrorOutput("Clusters section missing from kubeconfig file")
+
+        cluster = None
+        for cl in kubeconfig["clusters"]:
+            if "name" not in cl:
+                return "error", ErrorOutput(
+                    "'name' section missing from clusters entry in kubeconfig"
+                )
+            if cl["name"] == current_cluster:
+                if "cluster" not in cl:
+                    return "error", ErrorOutput(
+                        "cluster section missing from section "
+                        + "of current cluster {}".format(current_cluster)
+                    )
+                cluster = cl["cluster"]
+        if cluster is None:
+            return "error", ErrorOutput(
+                "Failed to find a cluster named {} in the kubeconfig file.".format(
+                    current_cluster
+                )
+            )
+
+        # Now find the user for current context's user for authentication.
+        if "users" not in kubeconfig:
+            return "error", ErrorOutput("'users' section not found in kubeconfig")
+        user = None
+        for u in kubeconfig["users"]:
+            for required_key in ["name", "user"]:
+                if required_key not in u:
+                    return "error", ErrorOutput(
+                        """'{0}' section in the users section not found in the """
+                        """kubeconfig""".format(required_key)
+                    )
+            if u["name"] == current_user:
+                user = u["user"]
+        if user is None:
+            return "error", ErrorOutput(
+                "Failed to find a user named {} in the kubeconfig file.".format(
+                    current_user
+                )
+            )
+
+        # Ensure the server is in the kubeconfig
+        if "server" not in cluster:
+            return "error", ErrorOutput(
+                """Failed to find server in cluster kubeconfig file {0}"""
+                """ cluster section""".format(current_cluster)
+            )
+        # Populate output values from the user and server sections.
+        output = SuccessOutput(
+            Connection(host=cluster["server"]),
+        )
+        output.connection.cacert = base64_decode(
+            cluster.get("certificate-authority-data", None)
+        )
+        output.connection.cert = base64_decode(
+            user.get("client-certificate-data", None)
+        )
+        output.connection.key = base64_decode(user.get("client-key-data", None))
+        output.connection.username = user.get("username", None)
+        output.connection.password = user.get("password", None)
+        output.connection.bearerToken = user.get("token", None)
+
+        return "success", output
+    except Exception as e:
+        # This is the catch-all case.
+        # The goal is for all other errors to be addressed individually.
+        print(traceback.format_exc())
```

Comment:
Why are we printing a traceback and returning an error?

If we do need to print an error, shouldn't it be on `stderr`?

Wouldn't we want to format the stack trace into the exception string returned?  Will the person using the plug-in be able to find the output printed to stdout separate from that returned by the plug-in?

Reply:
Added to the error output.